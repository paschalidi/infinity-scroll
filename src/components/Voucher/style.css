.Voucher {
    margin: 0 auto;
    max-width: 400px;
    margin-top: 10px;
    flex-direction: row;
    display: flex;
    -webkit-transition: all 0.3s ease-in-out;
    -moz-transition: all 0.3s ease-in-out;
    transition: all 0.3s ease-in-out;
    padding: 8px;
}

.Voucher:hover {
    transform: scale(1.01);
    background-color: white;
}

.Voucher-logo {
    width: 50%;
    display: flex;
}

.Voucher-logo img {
    width: 100%;
}

.Voucher-details {
    width: 50%;
    display: flex;
    flex-direction: column;
    height: inherit;
    padding-left: 8px;
}

.Voucher-price {
    font-size: 20px;
}

.Voucher-title {
    padding-top: 0.2vw;
}

.Voucher-desc {
    padding-top: 0.2vw;
    color: #c4c4c4;
    font-size: 14px;
}

.bottom {
    flex-grow: 1;
}
