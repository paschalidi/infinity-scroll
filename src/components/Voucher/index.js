/**
 *
 * Voucher
 *
 */
/* eslint-disable import/first */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */

import React from 'react';
import './style.css';
import Button from '../Button';
import PT from 'prop-types';


class Voucher extends React.Component { // eslint-disable-line react/prefer-stateless-function

  constructor(props) {
    super(props);
    this.state = { imgUrl: props.image };
  }

  handleImageOnError = () => this.setState({ imgUrl: 'http://via.placeholder.com/250x250' });

  render() {
    const {
            price,
            distribution,
            description,
            anchor
          } = this.props;
    const { imgUrl } = this.state;

    return (
      <div className='Voucher'>
        <div className='Voucher-logo'>
          <img src={imgUrl} alt={description} onError={() => this.props.handleImageOnError} />
        </div>
        <div className='Voucher-details'>
          <div className='Voucher-price'>Price: {price} €</div>
          <div className='Voucher-title'>{distribution}</div>
          <div className='Voucher-desc'>{description}</div>
          <Button onClick={() => console.log('You just bought a voucher!')}>
            {anchor}
          </Button>
        </div>
      </div>
    );
  }
}

Voucher.defaultProps = {
  image: 'https://fakeimg.pl/200x150/',
  price: 'Price: €100',
  distribution: 'Distributor',
  description: 'Description',
  anchor: 'Buy'
};

Voucher.propTypes = {
  image: PT.string.isRequired,
  price: PT.string.isRequired,
  distribution: PT.string.isRequired,
  description: PT.string.isRequired,
  anchor: PT.string.isRequired
};
export default Voucher;
