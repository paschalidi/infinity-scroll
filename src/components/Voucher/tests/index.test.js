import React from 'react';
import ReactDOM from 'react-dom';
import Voucher from 'index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render('<Voucher />', div);
  ReactDOM.unmountComponentAtNode(div);
});
