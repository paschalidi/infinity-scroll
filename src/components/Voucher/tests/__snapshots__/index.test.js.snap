// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voucher renders 1`] = `
<div
  className="Voucher"
>
  <div
    className="Voucher-logo"
  >
    <img
      alt="Description"
      onError={[Function]}
      src="https://fakeimg.pl/200x150/"
    />
  </div>
  <div
    className="Voucher-details"
  >
    <div
      className="Voucher-price"
    >
      Price: 
      Price: €100
       €
    </div>
    <div
      className="Voucher-title"
    >
      Distributor
    </div>
    <div
      className="Voucher-desc"
    >
      Description
    </div>
    <div
      className="bottom"
    />
    <div
      className="inline"
    >
      <button
        className="undefined btn"
        onClick={[Function]}
        style={undefined}
        type="button"
      >
        <style>
          
                      .btn {
                          letter-spacing: 0.10em;
                          outline: none;
                          overflow: hidden;
                          display: inline-block;
                          width: 100%;
                          font-size: 16px;
                          padding: 8px 15px;
                          margin: 0;
                          border-width: 0;
                          border-radius: 0;
                          color: white;
                          background-color: #199DAA;
                          transition: background-color .2s;
                          font-weight: 200;
                          cursor: pointer;
                      }
          
                      .btn:hover,
                      .btn:focus {
                          color: #f2f2f2;
                          background-color: #1AD6E8;
                      }
          
                      .btn.transparent {
                          background-color: transparent;
                          border: solid 1px #999;
                          color: #313541;
                      }
          
                      .btn.transparent:hover,
                      .btn.transparent:focus {
                          background-color: #999;
                          color: white;
                      }
                  
        </style>
        BUY
      </button>
    </div>
  </div>
</div>
`;
